src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+/**
+ * Exceptions encountered when interacting with a duke
+ */
+abstract class DukeException extends Exception {
+}
+
+/**
+ * Throws when the description of a task is empty.
+ */
+class EmptyDescriptionException extends DukeException {
+    private final String action;
+
+    public EmptyDescriptionException(String action) {
+        this.action = action;
+    }
+
+    @Override
+    public String toString() {
+        return "OOPS!!! The description of a " + this.action + " cannot be empty.\n";
+    }
+}
+
+/**
+ * Throws when the command is not recognised by duke.
+ */
+class UnrecognizableCommandException extends DukeException {
```

Comment:
Maybe can use enum for all the child class as a suggestion. Overall the code looks very good. Could not find any coding violations.